android/titanium/prebuild.js

```diff
@@ -0,0 +1,167 @@
+				return '\\u' + ('0000' + match.charCodeAt(0).toString(16)).slice(-4);
+			});
+		}
+		return stringValue;
+	}
+
+	// Fetch Titanium SDK's build version and git hash from environment variables.
+	let buildVersion = process.env.TITANIUM_SDK_BUILD_VERSION;
+	if (!buildVersion) {
+		buildVersion = '1.0.0';
+	}
+	let buildGitHash = process.env.TITANIUM_SDK_BUILD_GIT_HASH;
+	if (!buildGitHash) {
+		buildGitHash = 'HEAD';
+	}
+
+	// Create the "build.properties" content.
+	const fileContentString
+		= '#Generated by Titanium\n'
+		+ '\n'
+		+ 'build.version=' + escapeValue(buildVersion) + '\n'
+		+ 'build.githash=' + escapeValue(buildGitHash) + '\n'
+		+ 'build.timestamp=' + escapeValue(moment().format('YYYY/MM/DD HH:mm')) + '\n';
+
+	// Create the "build.properties" file under the "assets" directory.
+	const directoryPath = path.join(__dirname, 'assets', 'Resources', 'ti.internal');
+	const filePath = path.join(directoryPath, 'build.properties');
+	await fs.mkdirs(directoryPath);
+	return fs.writeFile(filePath, fileContentString);
+}
+
+/** Generates C/C++ source files containing internal JS files and from gperf templates. */
+async function generateSourceCode() {
+	// Generate a "KrollNativeBindings.h" file with perfect hashes via gperf tool.
+	// Note: 2nd argument is inserted into file as-is. Use relative path since absolute may contain user name.
+	const runtimeV8DirPath = path.join(__dirname, '..', 'runtime', 'v8');
+	await fs.mkdirs(path.join(runtimeV8DirPath, 'generated'));
+	await gperf(
+		runtimeV8DirPath, 'src/native/KrollNativeBindings.gperf',
+		path.join(runtimeV8DirPath, 'generated/KrollNativeBindings.h'));
+
+	// Create "bootstrap.js" which generates create*() functions for all Titanium internal modules.
+	// TODO: Replace our usage of python to make it easier to build on Windows.
+	await exec('python ' + quotePath(path.join(runtimeV8DirPath, 'tools', 'genBootstrap.py')) + ' --runtime=v8');
+
+	// Generate a "KrollGeneratedBindings.h" file with perfect hashes via gperf tool.
+	// Note: 2nd argument is inserted into file as-is. Use relative path since absolute may contain user name.
+	await gperf(
+		runtimeV8DirPath, 'generated/KrollGeneratedBindings.gperf',
+		path.join(runtimeV8DirPath, 'generated/KrollGeneratedBindings.h'));
+
+	// Fetch all JS file paths under directory: "./runtime/common/src/js"
+	const runtimeCommonDirPath = path.join(__dirname, '..', 'runtime', 'common');
+	let filePaths = await glob(
+		'*.js',
+		{
+			cwd: path.join(runtimeCommonDirPath, 'src', 'js'),
+			realpath: true
+		}
+	);
+
+	// Fetch all JS file paths under each module directory: "./modules/<ModuleName>/src/js"
+	filePaths = filePaths.concat(await glob(
+		'*/src/js/*.js',
+		{
+			cwd: path.join(__dirname, '..', 'modules'),
+			realpath: true
+		}
+	));
+
+	// Generate a "KrollJS.h" file containing all of the above fetched JS files.
+	// TODO: Replace our usage of python to make it easier to build on Windows.
+	let commandLine = 'python';
```

Comment:
Minor, but string manipulation could be reduced by doing something similar:
```JS
let args = [
	'python',
	quotePath(path.join(runtimeV8DirPath, 'tools', 'js2c.py')),
	quotePath(path.join(runtimeV8DirPath, 'generated', 'KrollJS.h')),
	quotePath(path.join(runtimeV8DirPath, 'generated', 'bootstrap.js'))
];
filePaths.forEach(path => args.push(quotePath(path)));

return exec(args.join(' '));
```

Reply:
I agree. Done.